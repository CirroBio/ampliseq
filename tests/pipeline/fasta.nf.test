nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_fasta"
    tag "pipeline"

    test("Fasta Input") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert snapshot(path("$outputDir/barrnap/rrna.arc.gff"),
                                path("$outputDir/barrnap/rrna.bac.gff"),
                                path("$outputDir/barrnap/rrna.euk.gff"),
                                path("$outputDir/barrnap/rrna.mito.gff")).match("barrnap") },
                { assert new File("$outputDir/barrnap/summary.tsv").exists() },
                { assert snapshot(path("$outputDir/dada2/ref_taxonomy.rdp_18.txt")).match("dada2") },
                { assert new File("$outputDir/dada2/ASV_tax_species.rdp_18.tsv").exists() },
                { assert new File("$outputDir/dada2/ASV_tax.rdp_18.tsv").exists() },
                { assert snapshot(path("$outputDir/input/ASV_seqs.fasta")).match("input") }
            )
        }
    }
}
