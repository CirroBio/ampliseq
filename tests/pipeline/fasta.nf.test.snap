{
    "input": {
        "content": [
            "ASV_seqs.fasta:md5,39b0d4ad96b37fa68d4ee5c21b667fed"
        ],
        "timestamp": "2023-05-28T21:06:17+0000"
    },
    "software_versions": {
        "content": [
            "{BARRNAP={barrnap=0.9}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.0, yaml=6.0}, DADA2_TAXONOMY={R=4.1.1, dada2=1.22.0}, Workflow={nf-core/ampliseq=2.6.0dev}}"
        ],
        "timestamp": "2023-05-28T21:06:17+0000"
    },
    "dada2": {
        "content": [
            "ref_taxonomy.rdp_18.txt:md5,815c4fce9f3d1de019fb995a43fb66ed"
        ],
        "timestamp": "2023-05-28T21:06:17+0000"
    },
    "barrnap": {
        "content": [
            "rrna.arc.gff:md5,6dae470aace9293d5eb8c318584852dd",
            "rrna.bac.gff:md5,439a9084f089120f700f938dfb58fa41",
            "rrna.euk.gff:md5,c9bc1d9d8fb77dc19c95dee2d53840eb",
            "rrna.mito.gff:md5,df19e1b84ba6f691d20c72b397c88abf"
        ],
        "timestamp": "2023-05-28T21:06:17+0000"
    }
}